docs/components/tree-list.md

```diff
@@ -0,0 +1,330 @@
+:::
+
+<KTreeList :items="defaultItems" />
+
+```html
+<KTreeList :items="items" />
+```
+
+## Props
+
+### items
+
+An array of items that make up the tree.
+
+<KTreeList :items="defaultItems2" />
+
+```html
+<template>
+  <KTreeList :items="items" />
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  data() {
+    return {
+      items: [{
+        name: "Cats",
+        id: 'cats'
+      },
+      {
+        name: "Dogs",
+        id: 'dogs',
+        children: [{
+          name: "Puppies",
+          id: 'puppies'
+        }]
+      },
+      {
+        name: "Bunnies",
+        id: 'bunnies'
+      }]
+    }
+  }
+})
+</script>
+```
+
+### disableDrag
+
+Boolean (defaults to `false`) to turn of drag-n-drop reordering of the list.
+
+<KTreeList disable-drag :items="disableItems" />
+
+```html
+<KTreeList disable-drag :items="items" />
+```
+
+### maxLevel
+
+Use this prop to customize the maximum supported depth of the tree. We default to a max level of `2` or support for parents, children, and grandchildren.
+
+<KTreeList :items="maxLevelItems" :max-level="1" />
+
+```html
+<KTreeList :items="items" :max-level="1" />
+```
+
+### v-model
+
+`KTreeList` works with v-model for data binding.
+
+::: danger
+You cannot use `v-model` with the `items` prop. You must use one or the other.
+:::
+
+<div>
+  <KLabel>Value:</KLabel> {{ myList }}
+  <KTreeList v-model="myList" />
+  <br>
+  <KButton @click="reset">Reset</KButton>
+</div>
+
+```html
+<template>
+  <KLabel>Value:</KLabel> {{ myList }}
+  <KTreeList v-model="myList" />
+  <KButton @click="reset">Reset</KButton>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  data() {
+    return {
+      myList: [{
+        name: "Cats",
+        id: 'cats'
+      },
+      {
+        name: "Dogs",
+        id: 'dogs',
+        children: [{
+          name: "Puppies",
+          id: 'puppies'
+        }]
+      },
+      {
+        name: "Bunnies",
+        id: 'bunnies'
+      }]
+    }
+  },
+  methods: {
+    reset() {
+      this.myList = [{
+        name: "Cats",
+        id: 'cats'
+      },
+      {
+        name: "Dogs",
+        id: 'dogs',
+        children: [{
+          name: "Puppies",
+          id: 'puppies'
+        }]
+      },
+      {
+        name: "Bunnies",
+        id: 'bunnies'
+      }]
+    }
+  }
+})
+</script>
+```
+
+## KTreeItem
+
+### Properties
+
+- `name` (required) - text displayed as the label for the item
+- `id` (required) - a unique value used to identify the item
+- `selected` - boolean to indicate whether the current item is selected or not
+- `icon` - `KIcon` name to be displayed to the left of the item `name` (defaults to `treeDoc`, specify `none` to not display any icon)
+- `children` - an array of `KTreeItem`'s that will be styled as children of the current item
```

Comment:
When you move this under the `items` prop, this would be changed to `an array of items that will be styled...`